dialogue-target/src/main/java/com/palantir/dialogue/RequestBody.java

```diff
@@ -31,6 +32,13 @@
     /** Returns <pre>true</pre> if {@link #writeTo(OutputStream)} may be invoked multiple times. */
     boolean repeatable();
 
+    /** Returns the {@code Content-Length} of the request body, if known. Note that there is no guarantee that
+     * this will result in a {@code Content-Length} request rather than {@code Transfer-Encoding: chunked}.
+     */
+    default OptionalLong length() {
+        return OptionalLong.empty();
+    }
```

Comment:
I thought we weren't intending to expose this in `:dialogue-target`, but just say if users happen to set a `Content-Length` header we might choose to respect it?

Reply:
I could go either way on this — the changes to this class are in an isolated commit.
We definitely aren’t including any changes to BinaryRequestBody, conjure consumers don’t interact with this interface though.

Reply:
I think for now I'd prefer to go with respecting the `Content-Length` header if the user provides it rather that extending the Request interface

Reply:
reverted that commit.